src/libav_streamer.cpp

```diff
@@ -282,41 +215,36 @@ void LibavStreamer::sendImage(const cv::Mat &img, const rclcpp::Time &time)
           (const uint8_t * const *)raw_frame->data, raw_frame->linesize, 0,
           output_height_, frame_->data, frame_->linesize);
 
-#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(55,28,1)
-  delete raw_frame;
-#else
   av_frame_free(&raw_frame);
-#endif
 
   // Encode the frame
-  AVPacket pkt;
-  int got_packet;
-  av_init_packet(&pkt);
-
-#if (LIBAVCODEC_VERSION_MAJOR < 54)
-  int buf_size = 6 * output_width_ * output_height_;
-  pkt.data = (uint8_t*)av_malloc(buf_size);
-  pkt.size = avcodec_encode_video(codec_context_, pkt.data, buf_size, frame_);
-  got_packet = pkt.size > 0;
-#elif (LIBAVCODEC_VERSION_MAJOR < 57)
-  pkt.data = NULL; // packet data will be allocated by the encoder
-  pkt.size = 0;
-  if (avcodec_encode_video2(codec_context_, &pkt, frame_, &got_packet) < 0)
+  AVPacket* pkt = av_packet_alloc();
+
+  ret = avcodec_send_frame(codec_context_, frame_);
+  if (ret == AVERROR_EOF)
   {
-     throw std::runtime_error("Error encoding video frame");
+    RCLCPP_DEBUG_STREAM(nh_->get_logger(), "avcodec_send_frame() encoder flushed\n"); 
   }
-#else
-  pkt.data = NULL; // packet data will be allocated by the encoder
-  pkt.size = 0;
-  if (avcodec_send_frame(codec_context_, frame_) < 0)
+  else if (ret == AVERROR(EAGAIN))
+  {
+    RCLCPP_DEBUG_STREAM(nh_->get_logger(), "avcodec_send_frame() need output read out\n"); 
+  }
+  if (ret < 0)
   {
     throw std::runtime_error("Error encoding video frame");
   }
-  if (avcodec_receive_packet(codec_context_, &pkt) < 0)
+
+  ret = avcodec_receive_packet(codec_context_, pkt);
+  bool got_packet = pkt->size > 0;
+  if (ret == AVERROR_EOF)
+  {
+    RCLCPP_DEBUG_STREAM(nh_->get_logger(), "avcodec_receive_packet() encoder flushed\n"); 
+  }
+  else if (ret == AVERROR(EAGAIN))
   {
-    throw std::runtime_error("Error retrieving encoded packet");
+    RCLCPP_DEBUG_STREAM(nh_->get_logger(), "avcodec_receive_packet() needs more input\n"); 
+    got_packet = false;
```

Comment:
Did you want to keep these debug logs before merging, or did you intend to remove them?

Reply:
The debug logs were backported from #103 . I don't see any reason to remove them.